src/View/Widgets/HeaderBar.vala

```diff
@@ -266,6 +266,31 @@ public class Files.View.Chrome.HeaderBar : Hdy.HeaderBar {
         });
 
         location_bar.escape.connect (() => {escape ();});
+
+        undo_manager = UndoManager.instance ();
+        undo_manager.request_menu_update.connect (set_undo_redo_tooltips);
+        set_undo_redo_tooltips ();
+    }
+
+    private void set_undo_redo_tooltips () {
+        unowned var undo_action_s = undo_manager.get_next_undo_description ();
+        unowned var redo_action_s = undo_manager.get_next_redo_description ();
+
+        undo_button.tooltip_markup = Granite.markup_accel_tooltip (
+            undo_accels,
+            undo_action_s != null ?
+            ///TRANSLATORS %s is a placeholder for a file operation type such as "Move"
+            _("Undo %s").printf (undo_action_s) :
```

Comment:
I think that we would have better quality by having a proper undo and redo sentence